include/mbgl/util/run_loop.hpp

```diff
@@ -64,14 +65,23 @@ class RunLoop : private util::noncopyable {
     template <class Fn, class Cb, class... Args>
     std::unique_ptr<WorkRequest>
     invokeWithCallback(Fn&& fn, Cb&& callback, Args&&... args) {
-        auto flag = std::make_shared<bool>();
+        auto flag = std::make_shared<std::atomic<bool>>();
```

Comment:
If it is just a flag, I would use `std::atomic_flag` which is guaranteed to be lock-free.